### PR TITLE
Add login

### PR DESCRIPTION
**Pull request by kristian240**

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -25,22 +26,23 @@ async function findService(dealId, headers) {
 }
 
 async function createConfig() {
-  const { token } = await inquirer.prompt([
-    { type: 'input', message: 'Productive.io token', name: 'token', type: 'password' },
-  ]);
+  const session = await createSession();
+
+  if (!session) {
```
Comment (andreicek) [inline src/config.js:31]: I would move this if into the createSession, this way we "box" the features and fail in the module.

File: src/config.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
 const fs = require('fs');
-const { get, fetchRemotePacakge } = require('./api');
+const { get, fetchRemotePacakge, post } = require('./api');
 const { promisify } = require('util');
 const inquirer = require('inquirer');
 const boxen = require('boxen');
 const packageJson = require('../package.json');
+const { createSession } = require('./session');
```
Comment (andreicek) [inline src/config.js:7]: If it's not an API I would really like if you could Namespace this import...

```
const Session = require('./session');
```

File: src/session.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    { 'Content-Type': 'application/vnd.api+json' }
+  );
+
+  if (!session.data) {
+    return null;
+  }
+
+  // check if 2FA is enabled
+  if (session.included[0].attributes.two_factor_auth) {
+    console.log('Seems like you have 2FA enabled. Please enter your 2FA code!');
```
Comment (andreicek) [inline src/session.js:22]: I would drop this log, it should be enough to just prompt the user.

File: src/session.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      },
+      { 'Content-Type': 'application/vnd.api+json' }
+    );
+
+    if (!sessionWith2FA.data) {
+      return null;
+    }
+  }
+
+  return session;
```
Comment (andreicek) [inline src/session.js:44]: Return only the data we need here not the whole JSON API response.

**Review by andreicek — CHANGES_REQUESTED** [review]
Thanks! This is amazing work...

Tokens seem to be unstable for some so this should be much better. We should also look into refreshing them but this is enough for now.

File: src/config.js (hunk 2 above)
Comment (kristian240) [reply src/config.js:7]: So in the code I would call `Session.create`?

File: src/config.js (hunk 1 above)
Comment (kristian240) [reply src/config.js:31]: How should I handle exception there? I guess instead of returning null, I could throw an error of some kind?

Comment (andreicek) [reply src/config.js:31]: Yeah, it's cli we just let it die. We'll worry about correct exit codes sometimes later.

**Review by andreicek — APPROVED** [review]

**Pull request merged**